t/URT/t/52_limit_cache_size.t

```diff
@@ -156,7 +161,7 @@ sub count_things_in_cache {
 #        foreach (values %{$UR::Context::all_objects_loaded->{$c}} ) {
 #            print "\tid ",$_->id,"\n";
 #        }
-        $count += scalar(values %{$UR::Context::all_objects_loaded->{$c}});
+        $count += scalar(grep { defined } values %{$UR::Context::all_objects_loaded->{$c}});
```

Comment:
Do you want to be counting the undefined ones too and checking that the cache has two items on line 29?